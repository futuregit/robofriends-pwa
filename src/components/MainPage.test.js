import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'Test@email.com'
        }],
        searchField: '',
        isPending: false
    };
    wrapper = shallow(<MainPage { ...mockProps }/>)
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'Test@email.com'
        }],
        searchField: 'john',
        isPending: false
    };
    const wrapper2 = shallow(<MainPage { ...mockProps2 }/>)
    expect(wrapper2.instance().filteredRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'Test@email.com'
    }])

})

it('filters robots correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'Test@email.com'
        }],
        searchField: 'a',
        isPending: false
    };
    const filteredRobots = []
    const wrapper3 = shallow(<MainPage { ...mockProps3 }/>)
    expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots)

})