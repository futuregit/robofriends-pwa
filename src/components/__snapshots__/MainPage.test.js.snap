// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={
          Array [
            Object {
              "email": "Test@email.com",
              "id": 3,
              "name": "John",
            },
          ]
        }
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
